sdk/go/common/resource/plugin/provider_plugin.go

```diff
@@ -181,6 +181,18 @@ func NewProvider(host Host, ctx *Context, pkg tokens.Package, version *semver.Ve
 		if jsonConfig != "" {
 			env = append(env, "PULUMI_CONFIG="+jsonConfig)
 		}
+
+		// TODO: where should this config come from?
+		projPath, err := workspace.DetectProjectPath()
+		if err == nil && projPath != "" {
+			project, err := workspace.LoadProject(projPath)
+			if err == nil {
+				env = append(env, fmt.Sprintf("PULUMI_PROJECT=%s", project.Name))
+				env = append(env, "PULUMI_STACK=%sbanana")
+				env = append(env, "PULUMI_ORGANIZATION=%stheorganizationgoeshere")
```

Comment:
I am not sure what the best approach is to thread this information through to here. [type defaultHost](https://github.com/pulumi/pulumi/blob/3bbd0c8a1bb4bb17ab4fc83f3f7385a9ffb98f43/sdk/go/common/resource/plugin/host.go#L232) already holds configuration, so I think we could add the stack & project names there as well.